src/core/drive/view_transitioner.ts

```diff
@@ -0,0 +1,31 @@
+declare global {
+  type ViewTransition = {
+    finished: Promise<void>
+  }
+
+  interface Document {
+    startViewTransition?(callback: () => Promise<void>): ViewTransition
+  }
+}
+
+export class ViewTransitioner {
+  private transitionStarted = false
+  private lastOperation = Promise.resolve()
+
+  renderChange(useTransition: boolean, render: () => Promise<void>) {
+    if (useTransition && this.canTransition && !this.transitionStarted) {
+      this.transitionStarted = true
+      this.lastOperation = this.lastOperation.then(async () => {
+        await document.startViewTransition!(render).finished
+      })
+    } else {
+      this.lastOperation = this.lastOperation.then(render)
+    }
+
+    return this.lastOperation
+  }
+
+  private get canTransition() {
```

Comment:
Maybe we can call this property `viewTransitionAvailable`? To be a bit more specific, there're a few different reasons why you can or can't use transitions, like meta tags.

Reply:
Good idea 👍